src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py

```diff
@@ -134,14 +134,14 @@ def load_command_table(self, _):
         g.keyvault_custom('add', 'add_certificate_issuer_admin')
         g.keyvault_custom('delete', 'delete_certificate_issuer_admin')
 
-    with self.command_group('keyvault storage', kv_data_sdk) as g:
-        g.keyvault_command('add', 'set_storage_account')
-        g.keyvault_command('list', 'get_storage_accounts')
-        g.keyvault_command('show', 'get_storage_account')
-        g.keyvault_command('update', 'update_storage_account')
-        g.keyvault_command('remove', 'delete_storage_account')
-        g.keyvault_command('regenerate-key', 'regenerate_storage_account_key')
-        if data_api_version != '2016_10_01':
+    if data_api_version != '2016_10_01':
+        with self.command_group('keyvault storage', kv_data_sdk) as g:
```

Comment:
This should use that standard syntax for specifying that commands only apply to a certain API version.

https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#keyword-argument-reference

Reply:
@tjprescott I tested setting min_api='7.0' for the command groups, however this results in the commands being filtered even when running on the latest profile.  Does the min_api parameter support filtering for our data-plane version? or only the management plane api version?

Reply:
min_api implementation expects the api version to be in date time format.. implementation here https://github.com/Azure/azure-cli/blob/ce4fac89bad98266ad6701e52c76771735775946/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L255

I think we should not have allowed 7.0 to be a version in the SDK level itself and standardize with date time format everywhere.

Reply:
@lmazuel implemented the change in CLI to support version comparisons between dates and arbitrary numbers. Laurent, can you take a look?

Reply:
If something needs to be fixed, it's the version comparison logic. We shouldn't allow this nonstandard way of filtering the commands.

Reply:
So, it goes deeper than I thought. I recommend to put max_api='42.0' for now (aka a value so high that's it's not a problem for the next century), and to discuss more deeply some design concern I have with Profile implemetation. Especially, the fact that profile name carries a semantic or not (actually they don't, but CLI expects they do and parse the name)